src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReferenceFinder.cs

```diff
@@ -171,7 +178,169 @@ private List<SymbolReference> DeDupeAndSortReferences(List<SymbolReference> allR
+                    }
+                    else
+                    {
+                        await HandleNugetReferenceAsync(
+                            installerService, allReferences, nameNode, project, isAttributeSearch, result, weight).ConfigureAwait(false);
+                    }
+
+                    weight++;
+                }
+            }
+
+            private async Task HandleReferenceAssemblyReferenceAsync(
+                IPackageInstallerService installerService,
+                List<Reference> allReferences,
+                TSimpleNameSyntax nameNode,
+                Project project,
+                bool isAttributeSearch,
+                PackageWithTypeResult result,
+                int weight,
+                CancellationToken cancellationToken)
+            {
+                foreach (var reference in project.MetadataReferences)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                    var assemblySymbol = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
+                    if (assemblySymbol?.Name == result.AssemblyName)
+                    {
+                        // Project already has a reference to an assembly with this name.
+                        return;
+                    }
+                }
+
+                var desiredName = GetDesiredName(isAttributeSearch, result);
+                allReferences.Add(new AssemblyReference(_owner, installerService,
+                    new SearchResult(desiredName, nameNode, result.ContainingNamespaceNames, weight), result.AssemblyName));
+            }
+
+            private Task HandleNugetReferenceAsync(
+                IPackageInstallerService installerService,
+                List<Reference> allReferences,
+                TSimpleNameSyntax nameNode,
+                Project project,
+                bool isAttributeSearch,
+                PackageWithTypeResult result,
+                int weight)
+            {
+                if (!installerService.IsInstalled(project.Solution.Workspace, project.Id, result.PackageName))
+                {
+                    var desiredName = GetDesiredName(isAttributeSearch, result);
+                    allReferences.Add(new PackageReference(_owner, installerService,
+                        new SearchResult(desiredName, nameNode, result.ContainingNamespaceNames, weight), result.PackageName));
+                }
+
+                return SpecializedTasks.EmptyTask;
+            }
+
+            private static string GetDesiredName(bool isAttributeSearch, PackageWithTypeResult result)
+            {
+                var desiredName = result.TypeName;
+                if (isAttributeSearch)
+                {
+                    desiredName = desiredName.GetWithoutAttributeSuffix(isCaseSensitive: false);
```

Comment:
because Meh :)

I have a bar which is like: wil this ever matter in practice?  If not... my caring level will be low.